src/context-selection/period-selector-bar-item/use-date-limit.test.js

```diff
@@ -0,0 +1,120 @@
+import getCurrentDate from '../../shared/fixed-periods/get-current-date.js'
+import { computeDateLimit } from './use-date-limit.js'
+
+jest.mock('../../shared/fixed-periods/get-current-date.js', () => ({
+    __esModule: true,
+    default: jest.fn(),
+}))
+
+/**
+ * Should add future periods, even when spanning over several years
+ */
+describe('computeDateLimit', () => {
+    afterEach(() => {
+        getCurrentDate.mockClear()
+    })
+
+    it('should be current date if no openFuturePeriods', () => {
+        getCurrentDate.mockImplementation(() => {
+            return new Date()
```

Comment:
I noticed in running these tests that they are a bit flakey with this implementation; the actual and expected get executed at slightly different times, so sometimes on my machine they fail the equality check as they differ by 1 ms. So, I think this test should be modified to return a constant time

Reply:
That's neat. Jan-Gerke actually added the getCurrentDate function with a note about not using it once jest.setSystemTime was working, so I wonder if he was running into problems with that before? I'm going to just return a constant from the mocked getCurrentDate for now, but will make a note to follow up if we should get rid of this function and just use jest.setSystemTime